copyrighter/copyrighter.py

```diff
@@ -0,0 +1,250 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2022 Megh Computing, Inc.
+
+# All rights reserved. No warranty, explicit or implied, provided.
+# Unauthorized use, modification, or distribution is strictly prohibited.
+
+import argparse
+import datetime
+import os
+import re
+import sys
+import typing
+
+# Constants.
+default_extensions = (".c", ".cpp", ".cs", ".css", ".h", ".hpp", ".java", ".js", ".php", ".py")
+
+# Globals.
+g_verbose = False
+
+
+def read_tombstone(file: typing.TextIO, max_lines: int = 9) -> str:
+    """Extract and return the tombstone from the file."""
+    lines = ""
+    for _ in range(max_lines):
+        line = file.readline()
+        if not line:
+            break
+        lines += line
+
+    return lines
+
+
+def read_file(filepath: str) -> list:
+    """Useful for unit testing."""
+    with open(filepath, "r") as file:
+        return file.readlines()
+
+
+def write_file(filepath: str, lines: list) -> None:
+    """Useful for unit testing."""
+    with open(filepath, "w") as file:
+        file.writelines(lines)
+
+
+def write_current_year(filepath: str, start_year: int, current_year: int):
+    lines = read_file(filepath)
+
+    # The subbing RE has to work for Megh and Apache.
+    copyright_regex = R" 20\d\d(-20\d\d)? Megh Computing, Inc."
+    copyright_replace = " {0}-{1} Megh Computing, Inc."
+
+    regex = re.compile(copyright_regex)
+    for i, line in enumerate(lines):
+        match = regex.search(line)
+        if match:
+            to_write = lines[:]
+            to_write[i] = regex.sub(copyright_replace.format(start_year, current_year), line)
+            # Write out.
+            write_file(filepath, to_write)
+            return
+    raise AssertionError()
+
+
+def parse_copyright_megh(lines: str) -> tuple:
+    copyright_regex_megh = R"Copyright \(c\) 20\d\d(-20\d\d)? Megh Computing, Inc."
+    match = re.search(copyright_regex_megh, lines)
+    if not match:
+        return None, None
+
+    start_year = int(match.group()[14:18])
```

Comment:
suggest adding a comment about how end_year will eq start_year if there is no (-dddd) in the line

Reply:
done